lib/portage/util/movefile.py

```diff
@@ -105,6 +105,35 @@ def _copyxattr(src, dest, exclude=None):
             )
 
 
+def _cmpxattr(src, dest, exclude=None):
+    """
+    Compares extended attributes between |src| and |dest| and returns True if xattrs
+    are not supported or if they are equal, False otherwise
+    """
+    try:
+        src_attrs = xattr.list(src)
+        dest_attrs = xattr.list(dest)
+    except OSError as e:
+        if e.errno != OperationNotSupported.errno:
+            raise
+        return True
+
+    if src_attrs:
+        if exclude is not None and isinstance(src_attrs[0], bytes):
+            exclude = exclude.encode(_encodings["fs"])
+    exclude = _get_xattr_excluder(exclude)
+
+    src_attrs = tuple(attr for attr in src_attrs if not exclude(attr))
+    dest_attrs = tuple(attr for attr in dest_attrs if not exclude(attr))
+    if src_attrs != dest_attrs:
```

Comment:
Is it possible for `xattr.list` to return the attr names in a different order? If so, it might be better to use a set instead of a tuple.

Reply:
Good point. I'm not sure if there exists a fixed order, but to be safe I changed it to a set with 81a3ec3

Reply:
I'm not seeing that change in the referenced commit or in this PR currently.

Reply:
Sorry, I was looking for the word "set" and missed that you used the more compact brace notation.